### PR TITLE
Add missing condition

### PR DESCRIPTION
**Pull request by pkliczewski**
Bug-Url: https://bugzilla.redhat.com/1884249
Signed-off-by: Piotr Kliczewski <piotr.kliczewski@gmail.com>

### CODE REVIEW

**Comment by pkliczewski** [comment]
CI failed with:
> Error while importing disk image: target-vm-attachment-1. pod CrashLoopBackoff restart exceeded

**Comment by pkliczewski** [comment]
Fixing disk size issue by updating reported disk size by fakeovirt:

> 'Unable to process data: Virtual image size 46137344 is larger than available size 43599790 (PVC size 46137344, reserved overhead 0.055000%). A larger PVC is required.'

**Comment by pkliczewski** [comment]
@mansam 2 vmware related tests failed. Please take a look.

File: pkg/controller/virtualmachineimport/virtualmachineimport_controller.go [hunk 1]
```diff
@@ -795,34 +795,33 @@ func setTrackerLabel(meta metav1.ObjectMeta, instance *v2vv1.VirtualMachineImpor
...
-	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			if err = r.updateProgress(instance, progressStartVM); err != nil {
+	if err != nil && k8serrors.IsNotFound(err) {
+		if err = r.updateProgress(instance, progressStartVM); err != nil {
+			return err
+		}
+		if err = r.updateToRunning(vmName); err != nil {
+			// Emit event vm failed to start:
+			r.recorder.Eventf(instance, corev1.EventTypeWarning, EventVMStartFailed, "Virtual Machine %s/%s failed to start: %s", vmName.Namespace, vmName.Name, err)
```
Comment (masayag) [inline pkg/controller/virtualmachineimport/virtualmachineimport_controller.go]: you can use `utils.ToLoggableResourceName(vmiName.Name, &vmiName.Namespace)`

File: pkg/controller/virtualmachineimport/virtualmachineimport_controller.go [hunk 2]
```diff
@@ -795,34 +795,33 @@ func setTrackerLabel(meta metav1.ObjectMeta, instance *v2vv1.VirtualMachineImpor
 func (r *ReconcileVirtualMachineImport) startVM(provider provider.Provider, instance *v2vv1.VirtualMachineImport, vmName types.NamespacedName) error {
 	vmi := &kubevirtv1.VirtualMachineInstance{}
 	err := r.client.Get(context.TODO(), vmName, vmi)
-	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			if err = r.updateProgress(instance, progressStartVM); err != nil {
+	if err != nil && k8serrors.IsNotFound(err) {
```
Comment (masayag) [inline pkg/controller/virtualmachineimport/virtualmachineimport_controller.go]: just for readability - I'd split this differently:
```
if err != nil {
  if k8serrors.IsNotFound(err) {
  ...
  } else {
    return err
  }
} else {
  // line 808-821
}
return nil
```

File: pkg/controller/virtualmachineimport/virtualmachineimport_controller.go [hunk 3]
```diff
@@ -795,34 +795,33 @@ func setTrackerLabel(meta metav1.ObjectMeta, instance *v2vv1.VirtualMachineImpor
...
+		}
+		if err = r.updateToRunning(vmName); err != nil {
+			// Emit event vm failed to start:
+			r.recorder.Eventf(instance, corev1.EventTypeWarning, EventVMStartFailed, "Virtual Machine %s/%s failed to start: %s", vmName.Namespace, vmName.Name, err)
+			return err
+		}
+	} else if err == nil {
+		if vmi.Status.Phase == kubevirtv1.Running || vmi.Status.Phase == kubevirtv1.Scheduled {
+			// Emit event vm is successfully imported and started:
+			r.recorder.Eventf(instance, corev1.EventTypeNormal, EventImportSucceeded, "Virtual Machine %s/%s imported and started", vmName.Namespace, vmName.Name)
```
Comment (masayag) [inline pkg/controller/virtualmachineimport/virtualmachineimport_controller.go]: you can use utils.ToLoggableResourceName(vmiName.Name, &vmiName.Namespace)

**Review by masayag — CHANGES_REQUESTED** [review]

File: pkg/controller/virtualmachineimport/virtualmachineimport_controller.go (hunk 2 above)
Comment (pkliczewski) [reply pkg/controller/virtualmachineimport/virtualmachineimport_controller.go]: Sure, we can do that.

File: pkg/controller/virtualmachineimport/virtualmachineimport_controller.go (hunk 1 above)
Comment (pkliczewski) [reply pkg/controller/virtualmachineimport/virtualmachineimport_controller.go]: sure

File: pkg/controller/virtualmachineimport/virtualmachineimport_controller.go (hunk 3 above)
Comment (pkliczewski) [reply pkg/controller/virtualmachineimport/virtualmachineimport_controller.go]: :+1:

**Review by masayag — APPROVED** [review]

**Comment by pkliczewski** [comment]
New issue in the CI:

> VirtualMachineImportCondition{type: Processing, status: False, reason: ProcessingFailed, message: Failed to find operating system for the VM} VirtualMachineImportCondition{type: Succeeded, status: False, reason: VMTemplateMatchingFailed, message: Couldn't find matching template. Either change the virtual machine OS type or add a custom template configMap for it, and a common template if there is none. Refer to documentation for more details.}

**Review by jakub-dzon — APPROVED** [review]

**Pull request merged**